src/kw/kw.cpp

```diff
@@ -371,11 +377,23 @@ bool KanoWorld::load_data() {
 
     const JSON_Object * const data = json_value_get_object(user_data.get());
 
-    this->set_token(json_object_get_string(data, "token"), false);
-    this->set_expiration_date(json_object_get_string(data, "duration"), false);
-    this->set_account_verified(
-        json_object_get_boolean(data, "is_verified"), false);
-    this->save_data();
+    if (json_object_dothas_value_of_type(data, "username", JSONString)) {
```

Comment:
Recommended to use `dothas` in conjunction with `"data.username"` and `has` with `"username"`.
```suggestion
    if (json_object_has_value_of_type(data, "username", JSONString)) {
```